src/MicrosoftTeams.ts

```diff
@@ -564,23 +746,64 @@ namespace microsoftTeams
             return;
         }
 
-        // Process only if the message is coming from a valid origin or the origin used for local testing
+        // Process only if the message is coming from a valid origin
+        let messageSource = evt.source || evt.originalEvent.source;
         let messageOrigin = evt.origin || evt.originalEvent.origin;
-        if (validOrigins.indexOf(messageOrigin.toLowerCase()) === -1)
+        if ((validOrigins.indexOf(messageOrigin.toLowerCase()) === -1) &&
+            (messageOrigin !== currentWindow.location.origin) ||
+            (messageSource === currentWindow))
         {
```

Comment:
Should these be grouped for clarity? Right now you have A && B || C but it'd be clearer to have A && (B || C)

Also a nit: You don't need the parens around each clause. You actually have (A) && (B) || (C), so those parens don't do anything.

So all in all, something like...
        if (validOrigins.indexOf(messageOrigin.toLowerCase()) === -1
            && (messageOrigin !== currentWindow.location.origin || messageSource === currentWindow))